### PR TITLE
[SPARK-39314][PS] Respect ps.concat sort parameter to follow pandas behavior

### PR DESCRIPTION
**Pull request by Yikun**
### What changes were proposed in this pull request?
Respect ps.concat sort parameter to follow pandas behavior:
- Remove the multi-index special sort process case and add ut.
- Still keep `num_series != 1` for now to follow pandas behavior

### Why are the changes needed?

Since pandas 1.4+ (https://github.com/pandas-dev/pandas/commit/01b8d2a77e5109adda2504b1cb4b1daeab3c74df),  ps.concat method the sort parameter. We need to follow pandas behavior.

### Does this PR introduce _any_ user-facing change?
Yes, but follow pandas 1.4 behavior

### How was this patch tested?
test_concat_index_axis, test_concat_multiindex_sort, concat doctest
passed with 1.3/1.4

### CODE REVIEW

File: python/pyspark/pandas/tests/test_namespace.py [hunk 1]
```diff
@@ -306,16 +329,30 @@ def test_concat_index_axis(self):
...
         objs = [
             ([psdf, psdf], [pdf, pdf]),
+            # no Series
             ([psdf, psdf.reset_index()], [pdf, pdf.reset_index()]),
             ([psdf.reset_index(), psdf], [pdf.reset_index(), pdf]),
             ([psdf, psdf[["C", "A"]]], [pdf, pdf[["C", "A"]]]),
             ([psdf[["C", "A"]], psdf], [pdf[["C", "A"]], pdf]),
+            # only one Series
             ([psdf, psdf["C"]], [pdf, pdf["C"]]),
+            # only one Series
```
Comment (itholic) [inline python/pyspark/pandas/tests/test_namespace.py:339]: nit: maybe we can remove this comment here ? The above one should cover.

**Review by itholic — COMMENTED** [review]
Looks good otherwise

File: python/pyspark/pandas/tests/test_namespace.py (hunk 1 above)
Comment (Yikun) [reply python/pyspark/pandas/tests/test_namespace.py:339]: Thanks! Resolved.

**Comment by Yikun** [comment]
also cc @HyukjinKwon @xinrong-databricks ready for review

**Comment by Yikun** [comment]
also cc @HyukjinKwon @xinrong-databricks ready for review

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**